cachito/web/api_v1.py

```diff
@@ -211,15 +212,25 @@ def download_archive(request_id):
         )
         raise InternalServerError()
 
+    hasher = hash_file(bundle_dir.bundle_archive_file)
+    checksum = hasher.hexdigest()
+    store_checksum = bundle_dir.bundle_archive_checksum.read_text(encoding="utf-8")
+    if checksum != store_checksum:
+        msg = "Checksum of bundle archive {} has changed."
+        flask.current_app.logger.error(msg.format(bundle_dir.bundle_archive_file))
+        raise InternalServerError(msg.format(bundle_dir.bundle_archive_file.name))
+
     flask.current_app.logger.debug(
         "Sending the bundle at %s for request %d", bundle_dir.bundle_archive_file, request_id
     )
-    return flask.send_file(
+    resp = flask.send_file(
         str(bundle_dir.bundle_archive_file),
         mimetype="application/gzip",
         as_attachment=True,
         attachment_filename=f"cachito-{request_id}.tar.gz",
     )
+    resp.headers["Digest"] = f"sha-256={hasher.base64digest()}"
```

Comment:
Would be safer to provide `store_checksum` here instead of dynamically computed hash?

Reply:
Not sure what's the point? :) Why would it be safer?

Reply:
if we change implementation in comparison and we wrongly compare values, passing original value will cause client to notice.

I prefer to send the static value directly not the computed one.

Reply:
Does "static value" mean the stored checksum?

Reply:
yes

Reply:
resolved f2f. We don't need this